src/raft.rs

```diff
@@ -1465,6 +1628,7 @@ impl<T: Storage> Raft<T> {
                 if pr.state == ProgressState::Replicate {
                     pr.become_probe();
                 }
+                self.groups.remove_delegate(m.from);
```

Comment:
What if it's delegate? We need to tell the leader the target becomes probe status.

Reply:
If delegates also handle peers' progress, it's ok.